packages/cli/src/config/schema.ts

```diff
@@ -917,12 +917,13 @@ export const schema = {
 		},
 	},
 
+	instanceRole: {
```

Comment:
why are we keeping this state in the config though? can this not be a property on one of the services instead?
we should stop using config to store data that user can't configure.

Reply:
I wouldn't feel comfortable making that change without [thorough retesting](https://github.com/n8n-io/n8n/pull/8446).

Reply:
This was likely done following what was done with `instanceType`. I think there's value in making `instanceRole` clearer until we decide to remove all state from the schema. Let me know what you prefer.

Reply:
does this make sense? https://github.com/n8n-io/n8n/compare/make-instance-role-clearer...10188-proposal?expand=1

Reply:
Thanks! In principle it does, what I'm nervous about is merging this without thorough testing.

Reply:
Let's bring it in. I'll do some spot checks testing tomorrow but I agree it looks safe.

Reply:
https://github.com/n8n-io/n8n/pull/10242